src/services/plan-service.ts

```diff
@@ -0,0 +1,129 @@
+import { DopplerLegacyClient } from './doppler-legacy-client';
+import {
+  FreePath,
+  StandardPath,
+  PlusPath,
+  AgenciesPath,
+  PlanType,
+  Plan,
+  PathType,
+} from '../doppler-types';
+
+export interface PlanHierarchy {
+  getPaths(userPlan: Plan): (FreePath | StandardPath | PlusPath | AgenciesPath)[];
+  // current plan free: FreePath, StandardPath, PlusPath, AgenciesPath
+  // current plan by credits: StandardPath, PlusPath, AgenciesPath
+  // current plan standard: StandardPath, PlusPath, AgenciesPath
+  // current plan plus: PlusPath, AgenciesPath
+
+  getPlanTypes(userPlan: Plan, pathType: PathType): PlanType[];
+  // get all plan types to be listed in slider: monthly-deliveries | subscribers | prepaid
+  // Free and select standard -->'PrepaidPack' | 'MonthlyRenewalDeliveriesPlan' | 'SubscribersLimitedPlan'
+
+  // Free and select plus --> 'MonthlyRenewalDeliveriesPlan' | 'SubscribersLimitedPlan'
+
+  // By credits -->  'PrepaidPack' | 'MonthlyRenewalDeliveriesPlan' | 'SubscribersLimitedPlan' (we should confirm this option)
+
+  // Standard and select standard (up contacts or up deliveries option) --> 'MonthlyRenewalDeliveriesPlan' | 'SubscribersLimitedPlan'
+
+  // Standard and select plus --> 'MonthlyRenewalDeliveriesPlan' | 'SubscribersLimitedPlan'
+
+  // plus and select plus (up contacts or up deliveries option) -->  'MonthlyRenewalDeliveriesPlan' | 'SubscribersLimitedPlan'
+
+  getPlans(userPlan: Plan, pathType: PathType, planType: PlanType): Plan[];
+  // this method is to actually get a plans array with all options to show in slider, in this method we must ensure to call BE once
+}
+
+export class PlanService implements PlanHierarchy {
+  private PlanList: Plan[] = [];
+  private readonly dopplerLegacyClient: DopplerLegacyClient;
+
+  constructor({ dopplerLegacyClient }: { dopplerLegacyClient: DopplerLegacyClient }) {
+    this.dopplerLegacyClient = dopplerLegacyClient;
+  }
+
+  async ensurePlanListLoaded(): Promise<Plan[]> {
+    return this.PlanList.length
+      ? this.PlanList
+      : (this.PlanList = await this.dopplerLegacyClient.getAllPlans());
+  }
+
+  private getMinimumFee(path:PathType):number{
+    throw new Error('Not implemented');
+  }
+
+  private isHigherPlan(plan:Plan):boolean {
+    throw new Error('Not implemented');
+  }
+
+  private getMinimumFeePrepaid(): number {
+    throw new Error('Not implemented');
+  }
+
+  getPaths(userPlan: Plan): (FreePath | StandardPath | PlusPath | AgenciesPath)[] {
```

Comment:
@andresmoschini I don´t like this, but I don´t figure a way out of this, what do you think?

Reply:
I think that it is fine, but you could define a new type `Path` with the union:

```typescript
type Path = FreePath | StandardPath | PlusPath | AgenciesPath;

getPaths(userPlan: Plan): Path[] {
```

Reply:
I was refering to the function anyway... with the switch and so many objects ... :trollface:

Reply:
Oh, I understand now haha.

I will review it during the day.